crates/rpc/rpc-api/src/optimism.rs

```diff
@@ -135,22 +155,33 @@ pub struct PeerScores {
     pub req_resp: ReqRespScores,
 }
 
+/// https://github.com/ethereum-optimism/optimism/blob/develop/op-node/p2p/rpc_api.go#L15
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct PeerInfo {
+    #[serde(rename = "peerID")]
     pub peer_id: String,
+    #[serde(rename = "nodeID")]
     pub node_id: String,
     pub user_agent: String,
     pub protocol_version: String,
+    #[serde(rename = "ENR")]
     pub enr: String,
     pub addresses: Vec<String>,
-    pub protocols: Vec<String>,
-    pub connectedness: PeerConnectedness,
-    pub direction: Direction,
+    pub protocols: Option<Vec<String>>,
+    /// 0: "NotConnected", 1: "Connected",
+    /// 2: "CanConnect" (gracefully disconnected)
+    /// 3: "CannotConnect" (tried but failed)
+    pub connectedness: u8,
+    /// 0: "Unknown", 1: "Inbound" (if the peer contacted us)
+    /// 2: "Outbound" (if we connected to them)
+    pub direction: u8,
     pub protected: bool,
+    #[serde(rename = "chainID")]
     pub chain_id: ChainId,
-    pub latency: Duration,
+    pub latency: u64,
```

Comment:
is this seconds or ms?

Reply:
time.duration in go, it should be nanosecond.